### PR TITLE
Disable all image hardware decoding in the WebContent process

### PR DESCRIPTION
**Pull request by pvollan**
#### c0fc79a6a65c4004c014a4f8556720d2bbc1ad48
<pre>
Disable all image hardware decoding in the WebContent process
<a href="https://bugs.webkit.org/show_bug.cgi?id=259457">https://bugs.webkit.org/show_bug.cgi?id=259457</a>
rdar://112477703

Reviewed by Brent Fulgham.

Disable hardware decoding for all image formats in the WebContent process when IOKit is blocked.

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/PAL/pal/spi/cg/ImageIOSPI.h:
* Source/WebKit/WebProcess/WebPage/WebPage.cpp:
* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::WebProcess::platformInitializeWebProcess):

Canonical link: <a href="https://commits.webkit.org/266335@main">https://commits.webkit.org/266335@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/4bae5305ca74efe5672ec0ce553f37bc5639b618

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/13561 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/13875 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/14206 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/15298 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/12897 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/13640 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/16383 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/13953 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/15580 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/13728 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/14371 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/11481 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/16002 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/11659 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/12236 "Passed tests") | [⏳ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/GTK-WK2-Tests-EWS "Waiting to run tests") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/11558 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/12734 "Passed tests") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/12404 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/15618 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/12842 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/12922 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/10804 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/13605 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/12190 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/3552 "Passed tests") | 
| [❌ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/3302 "Failed to push commit to Webkit repository") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/16518 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/13990 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/12765 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/3353 "Passed tests") |

### CODE REVIEW

**Review by brentfulgham — APPROVED** [review]
LGTM r=me.

File: Source/WTF/wtf/PlatformHave.h [hunk 1]
```diff
@@ -1485,3 +1485,7 @@
     || (PLATFORM(WATCHOS) && __WATCH_OS_VERSION_MIN_REQUIRED >= 100000)
 #define HAVE_CTFONTCOPYCOLORGLYPHCOVERAGE 1
 #endif
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000)
```
Comment (smfr) [inline Source/WTF/wtf/PlatformHave.h]: What about other Apple platforms?

**Comment by pvollan** [comment]
> LGTM r=me.

Thanks for reviewing!

File: Source/WTF/wtf/PlatformHave.h (hunk 1 above)
Comment (pvollan) [reply Source/WTF/wtf/PlatformHave.h]: > What about other Apple platforms?

Ah, good point! I initially thought we'd keep this change macOS only, but it is good to align the platforms. Fixed in latest patch.

Thanks for reviewing!

File: Source/WTF/wtf/PlatformHave.h [hunk 2]
```diff
@@ -1488,3 +1488,11 @@
 #if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000)
 #define HAVE_NSWINDOW_SNAPSHOT_READINESS_HANDLER 1
 #endif
+
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
```
Comment (aproskuryakov) [inline Source/WTF/wtf/PlatformHave.h]: Doesn't `PLATFORM(MAC)` include macCatalyst?

Also, what about visionOS?

**Review by aproskuryakov — COMMENTED** [review]
Could you please expand the title of the change (and of the bug) to include which hardware decoding is being disabled, and where? It is currently extremely confusing between multiple processes, and also various video and image formats.

As far as I can tell, this is abut all image formats in WebContent?

File: Source/WTF/wtf/PlatformHave.h (hunk 2 above)
Comment (brentfulgham) [reply Source/WTF/wtf/PlatformHave.h]: I believe `PLATFORM(IOS_FAMILY)` includes macCatalyst, but plain `PLATFORM(MAC)` does not. I'm not sure if `OS(VISION)` is part of `IOS_FAMILY` yet.

Comment (aproskuryakov) [reply Source/WTF/wtf/PlatformHave.h]: I checked the headers yesterday, and I believe that both do. But it's complicated, and I could easily get it wrong without testing.

**Comment by pvollan** [comment]
> Could you please expand the title of the change (and of the bug) to include which hardware decoding is being disabled, and where? It is currently extremely confusing between multiple processes, and also various video and image formats.
> 
> As far as I can tell, this is abut all image formats in WebContent?

Ah, yes, that is a good point. I will update the patch with more detailed information.

Thanks for reviewing!

File: Source/WTF/wtf/PlatformHave.h [hunk 3]
```diff
@@ -1469,6 +1469,15 @@
 #define HAVE_NSVIEW_CLIPSTOBOUNDS_API 1
 #endif
 
+#if (PLATFORM(MAC) && __MAC_OS_X_VERSION_MIN_REQUIRED >= 140000) \
+    || (PLATFORM(MACCATALYST) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
+    || (PLATFORM(IOS) && __IPHONE_OS_VERSION_MIN_REQUIRED >= 170000) \
```
Comment (brentfulgham) [inline Source/WTF/wtf/PlatformHave.h:1473]: Do we need PLATFORM(VISION)? Should it be PLATFORM(IOS_FAMILY)?

**Review by brentfulgham — APPROVED** [review]

File: Source/WTF/wtf/PlatformHave.h (hunk 2 above)
Comment (pvollan) [reply Source/WTF/wtf/PlatformHave.h]: Yes, I believe PLATFORM(IOS_FAMILY) includes macCatalyst, but PLATFORM(MAC) does not. This is not required for visionOS at the moment, I believe.

File: Source/WTF/wtf/PlatformHave.h (hunk 3 above)
Comment (pvollan) [reply Source/WTF/wtf/PlatformHave.h:1473]: This is not required for visionOS at the moment, I believe.

Thanks for reviewing!

File: Source/WTF/wtf/PlatformHave.h (hunk 2 above)
Comment (aproskuryakov) [reply Source/WTF/wtf/PlatformHave.h]: > Yes, I believe PLATFORM(IOS_FAMILY) includes macCatalyst, but PLATFORM(MAC) does not.

Indeed.

**Pull request merged**